### PR TITLE
perf: cache nested import of get_user_or_annonymous to improve performance

### PR DESCRIPTION
**Pull request by bellini666**

### CODE REVIEW

File: strawberry_django/permissions.py [hunk 1]
```diff
@@ -48,18 +48,28 @@
 from .utils.query import filter_for_user
 from .utils.typing import UserType
 
-try:
-    # Try to use the smaller/faster cache decorator if available
-    _cache = functools.cache  # type: ignore
-except AttributeError:
-    _cache = functools.lru_cache
-
```
Comment (bellini666) [inline strawberry_django/permissions.py:56]: this was not being used lol

File: strawberry_django/permissions.py [hunk 2]
```diff
@@ -48,18 +48,28 @@
...
 
 
 _M = TypeVar("_M", bound=Model)
 
+_sentinel: Any = object()
+_user_or_anonymous_getter: Optional[Callable[[UserType], UserType]] = _sentinel
+
+
+def _get_user_or_anonymous_getter() -> Optional[Callable[[UserType], UserType]]:
+    global _user_or_anonymous_getter  # noqa: PLW0603
```
Comment (patrick91) [inline strawberry_django/permissions.py:62]: wouldn't this be nicer with the lru cache?

Comment (bellini666) [reply strawberry_django/permissions.py:62]: Wasn't sure if that was an overkill, but you are probably right! Going to change it

Comment (bellini666) [reply strawberry_django/permissions.py]: Done!

**Review by patrick91 — APPROVED** [review]

**Pull request merged**